src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvocationContext.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Shared.Collections;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Provides context for an in-flight function invocation.</summary>
+public sealed class FunctionInvocationContext
```

Comment:
I take it this change is based on top of the other PR branch you have open atm?

Reply:
Yes, it's mentioned in the PR description.

> Ignore the first commit, it's just https://github.com/dotnet/extensions/pull/5954.